search.go

```diff
@@ -63,12 +63,50 @@ func (req *SearchRequest) Size(size uint64) *SearchRequest {
 	return req
 }
 
-// Sort sets how the results should be sorted.
-func (req *SearchRequest) Sort(params ...SortParams) *SearchRequest {
-	req.sort = params
+// Sort appends one or more sort options.
+// Accepts any type that implements SortOption (field, script, raw)
+func (req *SearchRequest) Sort(opts ...SortOption) *SearchRequest {
+	if opts != nil {
+		req.sort = append(req.sort, opts...)
+	}
+	return req
+}
+
+// SortScript is a convenience method for script-based sorting
+func (req *SearchRequest) SortScript(params ScriptSortParams) *SearchRequest {
+	req.sort = append(req.sort, params)
+	return req
+}
+
+// SortRaw allows sorting by a raw string like "_score"
+func (req *SearchRequest) SortRaw(field string) *SearchRequest {
+	if field == "" {
+		return req
+	}
+	req.sort = append(req.sort, rawSortField(field))
 	return req
 }
 
+// ClearSort removes all existing sort options
+func (req *SearchRequest) ClearSort() *SearchRequest {
+	req.sort = req.sort[:0]
+	return req
+}
+
+// SortByScript creates a new script-based sort option
+func (req *SearchRequest) SortByScript(scriptField *ScriptField, sortType string, order Order) *SearchRequest {
+	if scriptField == nil {
+		return req
+	}
+
+	params := ScriptSortParams{
```

Comment:
this should not be done here ScriptSortParams should have setter functions and this function should not be required.